src/python/py/models/builder.py

```diff
@@ -576,11 +580,195 @@
 
             self.rotemb_attrs["create_rotary_embedding_caches"] = False
 
+        return cos_cache_name, sin_cache_name
+
+    def make_rotary_embedding(self, rotemb, name, root_input, **kwargs):
+        cos_cache_name, sin_cache_name = self.make_rotary_embedding_caches(rotemb)
+
         inputs = [root_input, kwargs.pop("position_ids"), cos_cache_name, sin_cache_name]
         output = f"{name}/output_0"
         self.make_node("RotaryEmbedding", inputs=inputs, outputs=[output], name=name, domain="com.microsoft", interleaved=0, **kwargs)
         self.make_value_info(output, self.io_dtype, shape=['batch_size', 'sequence_length', self.head_size * (self.num_kv_heads if "k_rotary" in name else self.num_attn_heads)])
 
+    # TODO: This function and any corresponding changes to support it are temporary until ORT supports GQA for CPU
+    def make_repeat_kv(self, layer_id, root_input, past_kv, present_kv, **kwargs):
+        # Make subgraph that repeats tensor of shape (batch_size, sequence_length, num_kv_heads, head_size)
+        # to shape (batch_size, sequence_length, num_attn_heads, head_size) in an interleaved pattern
+        # and updates the KV caches
+        #
+        #           root_input
+        #                |
+        #             Reshape
+        #                |
+        #            Transpose
```

Comment:
>Transpose [](http://example.com/codeflow?start=21&length=9)

Why it needs Transpose? Transpose is quite expensive

Reply:
```
                                        past_kv (BxNxSxH)
                                           |
root_input --> Reshape --> Transpose --> Concat (BxNxSxH)
  (BxSxD)     (BxSxNxH)    (BxNxSxH)       |
                                       present_kv (BxNxSxH)
```
The root input for the K path is the output from `RotaryEmbedding` and the output is of shape `BxSxD`. The root input for the V path is the output from `MatMul` and the output is of shape `BxSxD`. Thus, the root input to `repeat_kv` is of shape `BxSxD` in both cases.

The past key/past value and present key/present value are each of shape `BxNxSxH`. In order to concatenate the past key/past value with the current key/current value, the current key/current value has to be of shape `BxNxSxH`. The `Reshape` converts the root input to shape `BxSxNxH` and the `Transpose` converts it to shape `BxNxSxH`. Hence, the `Transpose` is needed.